p2p/host/basic/basic_host.go

```diff
@@ -104,6 +104,7 @@ func New(net inet.Network, opts ...interface{}) *BasicHost {
 
 // newConnHandler is the remote-opened conn handler for inet.Network
 func (h *BasicHost) newConnHandler(c inet.Conn) {
+	h.Peerstore().SetProtocols(c.RemotePeer())
```

Comment:
Do you want to put it in comment in there? It is not really clear from the code.